### PR TITLE
APPS-1283 Introduce DTO layer

### PR DESCRIPTION
**Pull request by korotkov-aerospike**

### CODE REVIEW

**Comment by korotkov-aerospike** [comment]
Brief overview:

1) split dto/model
 all validation in dto
 logic in model
 dtos have to/From model methods
 serialization for dto.

2) config validation in model. Config is always valid, it's checked on add/remove elements
3) configuration manager moved to internal package
4) use actual objects instead of names in business logic, e.g. ClientManager now stores pointers to actual cluster structs.

all tests pass: http://3.65.168.247:8080/ui/#backup/launches/all

File: internal/server/configuration/configuration_manager_local.go [hunk 1]
```diff
@@ -1,51 +1,46 @@
...
 		return nil, errors.New("configuration file is missing")
 	}
-	buf, err := os.ReadFile(filePath)
-	if err != nil {
-		return nil, err
-	}
 
-	config := model.NewConfigWithDefaultValues()
-	err = yaml.Unmarshal(buf, config)
+	file, err := os.Open(filePath)
```
Comment (reugn) [inline internal/server/configuration/configuration_manager_local.go]: Just
```go
return os.Open(filePath)
```

File: pkg/dto/backup_routine.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+		return emptyFieldValidationError("backup policy")
+	}
+	if r.SourceCluster == "" {
+		return emptyFieldValidationError("source-cluster")
+	}
+	if r.Storage == "" {
+		return emptyFieldValidationError("storage")
+	}
+	if err := quartz.ValidateCronExpression(r.IntervalCron); err != nil {
+		return fmt.Errorf("backup interval string '%s' invalid: %v", r.IntervalCron, err)
```
Comment (reugn) [inline pkg/dto/backup_routine.go]: %v -> %w

File: pkg/dto/backup_routine.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	}
+	if r.IncrIntervalCron != "" { // incremental interval is optional
+		if err := quartz.ValidateCronExpression(r.IncrIntervalCron); err != nil {
+			return fmt.Errorf("incremental backup interval string '%s' invalid: %v", r.IntervalCron, err)
```
Comment (reugn) [inline pkg/dto/backup_routine.go]: %v -> %w

File: pkg/dto/config.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	return &Config{
+		ServiceConfig:     NewBackupServiceConfigWithDefaultValues(),
+		Storage:           map[string]*Storage{},
+		BackupRoutines:    map[string]*BackupRoutine{},
+		BackupPolicies:    map[string]*BackupPolicy{},
+		AerospikeClusters: map[string]*AerospikeCluster{},
+	}
+}
+
+func NewConfigFromModel(m *model.Config) *Config {
```
Comment (reugn) [inline pkg/dto/config.go:63]: Check `m` for nil to be consistent.

File: pkg/dto/backup_policy.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+		Bandwidth:         p.Bandwidth,
+		RecordsPerSecond:  p.RecordsPerSecond,
+		FileLimit:         p.FileLimit,
+		EncryptionPolicy:  p.EncryptionPolicy.ToModel(),
+		CompressionPolicy: p.CompressionPolicy.ToModel(),
+		Sealed:            p.Sealed,
+	}
+}
+
+func NewBackupPolicyFromModel(m *model.BackupPolicy) *BackupPolicy {
```
Comment (reugn) [inline pkg/dto/backup_policy.go:140]: Check `m` for nil to be consistent.

File: pkg/dto/backup_details.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	d.Namespace = m.Namespace
+	d.RecordCount = m.RecordCount
+	d.ByteCount = m.ByteCount
+	d.FileCount = m.FileCount
+	d.SecondaryIndexCount = m.SecondaryIndexCount
+	d.UDFCount = m.UDFCount
+}
+
+// NewBackupDetailsFromModel creates a new BackupDetails from a model.BackupDetails
+func NewBackupDetailsFromModel(m *model.BackupDetails) *BackupDetails {
```
Comment (reugn) [inline pkg/dto/backup_details.go:52]: Check `m` for nil to be consistent.

File: pkg/dto/aerospike_cluster.go [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+	}
+
+	if err := a.Validate(); err != nil {
+		return nil, err
+	}
+
+	return a, nil
+}
+
+func NewClusterFromModel(m *model.AerospikeCluster) *AerospikeCluster {
```
Comment (reugn) [inline pkg/dto/aerospike_cluster.go:69]: Check `m` for nil to be consistent.

File: pkg/dto/restore_namespace.go [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+// @Description operation.
+type RestoreNamespace struct {
+	// Original namespace name.
+	Source *string `json:"source,omitempty" example:"source-ns" validate:"required"`
+	// Destination namespace name.
+	Destination *string `json:"destination,omitempty" example:"destination-ns" validate:"required"`
+}
+
+// Validate validates the restore namespace.
+func (n *RestoreNamespace) Validate() error {
```
Comment (reugn) [inline pkg/dto/restore_namespace.go:23]: Check `n` for nil.

File: pkg/dto/secret_agent.go [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+		ConnectionType: s.ConnectionType,
+		Address:        s.Address,
+		Port:           s.Port,
+		Timeout:        s.Timeout,
+		TLSCAString:    s.TLSCAString,
+		IsBase64:       s.IsBase64,
+	}
+}
+
+func NewSecretAgentFromModel(m *model.SecretAgent) *SecretAgent {
```
Comment (reugn) [inline pkg/dto/secret_agent.go:42]: Check `m`for nil.

**Review by reugn — APPROVED** [review]

**Pull request merged**